packages/rollup-plugin-html/src/getInputHtmlData.js

```diff
@@ -1,39 +1,107 @@
 /** @typedef {import('rollup').InputOptions} InputOptions */
 /** @typedef {import('./types').PluginOptions} PluginOptions */
-/** @typedef {import('./types').InputHtmlData} InputHtmlData */
+/** @typedef {import('./types').HtmlFile} HtmlFile */
 
 const fs = require('fs-extra');
 const path = require('path');
+const glob = require('glob');
 const { createError } = require('./utils');
 
+/**
+ * Lists all files using the specified glob, starting from the given root directory.
+ *
+ * Will return all matching file paths fully resolved.
+ * @param {string} fromGlob
+ * @param {string} rootDir
+ * @return {string[]}
+ */
+function listFiles(fromGlob, rootDir) {
+  const files = glob.sync(fromGlob, { cwd: rootDir, absolute: true });
+  return (
+    files
+      // .map(filePath => path.resolve(rootDir, filePath))
+      // filter out directories
+      .filter(filePath => !fs.lstatSync(filePath).isDirectory())
+  );
+}
+
+/**
+ * @param {string} globs
+ * @param {string} rootDir
+ */
+function globsToFiles(globs, rootDir) {
+  const patterns = typeof globs === 'string' ? [globs] : globs;
+
+  /** @type {string[]} */
+  let files = [];
+  for (const pattern of patterns) {
+    files = [...files, ...listFiles(pattern, rootDir)];
+  }
+  return files;
+}
+
+/**
+ * @param {string} filePath
+ * @param {string} rootDir
+ * @param {boolean} flatten
+ */
+function getName(filePath, rootDir, flatten = true) {
+  if (flatten) {
+    return path.basename(filePath);
+  }
+  return filePath.substring(rootDir.length + 1);
```

Comment:
I think path.relative is safer here cross platform

Reply:
👍